CondFormats/L1TObjects/src/L1TUtmTriggerMenu.cc

```diff
@@ -1,2 +1,67 @@
 // auto-generated file by import_utm.pl
 #include "CondFormats/L1TObjects/interface/L1TUtmTriggerMenu.h"
+
+//copied from tmUtil
+const unsigned long L1TUtmTriggerMenu::getFirmwareUuidHashed() const {
+  const void* key = getFirmwareUuid().c_str();
+  int len = getFirmwareUuid().size();
+  unsigned int seed = 3735927486;
+  return murmurHashNeutral2(key, len, seed);
+}
+
+//copied from tmUtil
+unsigned long L1TUtmTriggerMenu::murmurHashNeutral2(const void* key, int len, unsigned int seed) {
+  // 'm' and 'r' are mixing constants generated offline.
+  // They're not really 'magic', they just happen to work well.
+
+  const unsigned int m = 0x5bd1e995;
+  const int r = 24;
+
+  // Initialize the hash to a 'random' value
+
+  unsigned int h = seed ^ len;
+
+  // Mix 4 bytes at a time into the hash
+
+  const unsigned char* data = (const unsigned char*)key;
+
+  while (len >= 4) {
+    unsigned int k;
+
+    k = data[0];
+    k |= data[1] << 8;
+    k |= data[2] << 16;
+    k |= data[3] << 24;
+
+    k *= m;
+    k ^= k >> r;
+    k *= m;
+
+    h *= m;
+    h ^= k;
+
+    data += 4;
+    len -= 4;
+  }
+
+  // Handle the last few bytes of the input array
+
+  switch (len) {
+    case 3:
+      h ^= data[2] << 16;
+    case 2:
+      h ^= data[1] << 8;
+    case 1:
```

Comment:
@Sam-Harper Is the purpose of `case 3:` and `case 2:` to fall through (i.e. is the `break;` omitted on purpose)? 

(spotted by gcc 9 warnings)

Reply:
no idea. This is was copied from https://gitlab.cern.ch/cms-l1t-utm/utm/-/blob/master/tmUtil/tmUtil.cc

Reply:
Do you know who would know?

Reply:
no, @rekovic might know though. Here we are emulating firmware deployed at p5 though. So even if it is "buggy" its still correct I think as long as it agrees with the hardware output at p5.

Reply:
@makortel @Sam-Harper , looks like the fall through is on purpose. I have opened a PR https://github.com/cms-sw/cmssw/pull/31828 to silence GCC 9 warnings